frontend/public/components/row-filters.tsx

```diff
@@ -0,0 +1,349 @@
+      _.forIn(filters, (value, key) => {
+        const recognized = _.filter(selected, (item) => value.includes(item));
+        if (recognized.length > 0) {
+          setQueryArgument(filterKeys[key], recognized.join(','));
+        } else {
+          removeQueryArgument(filterKeys[key]);
+        }
+      });
+    }
+  };
+
+  const updateRowFilterSelected = (id) => {
+    const selectedNew = _.xor(selectedRowFilters, id);
+    applyRowFilter(selectedNew);
+    setQueryParameters(selectedNew);
+    setOpen(false);
+  };
+
+  const clearAllRowFilter = (f) => {
+    updateRowFilterSelected(_.intersection(filters[f], selectedRowFilters));
+  };
+
+  const onRowFilterSelect = (event) => {
+    event.preventDefault();
+    updateRowFilterSelected([event?.target?.id]);
+  };
+
+  const clearAll = () => {
+    Object.keys(filters).forEach((key) => clearAllRowFilter(key));
+    if (!hideNameFilter) {
+      updateNameFilter('');
+    }
+    if (!hideLabelFilter) {
+      updateLabelFilter([]);
+    }
+  };
+
+  // Initial URL parsing
+  React.useEffect(() => {
+    !_.isEmpty(labelFilters) && applyFilter(labelFilters, FilterType.LABEL);
+    !_.isEmpty(nameFilter) && applyFilter(nameFilter, FilterType.NAME);
+    !_.isEmpty(selectedRowFilters) && applyRowFilter(selectedRowFilters);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  const switchFilter = (type: FilterType) => {
+    setFilterType(FilterType[type]);
+    setInputText('');
+  };
+
+  const dropdownItems = getDropdownItems(rowFilters, selectedRowFilters, data);
+
+  return (
+    <>
+      <div className="co-search-group co-search-group__filter co-filter__block">
+        {rowFilters.length > 0 && (
+          <div className="co-type-selector">
+            <Dropdown
+              dropdownItems={dropdownItems}
+              className="co-list-filter-block__row-dropdown co-search-group__resource"
+              onSelect={onRowFilterSelect}
+              isOpen={isOpen}
+              toggle={
+                <DropdownToggle onToggle={() => setOpen(!isOpen)} iconComponent={CaretDownIcon}>
+                  <FilterIcon className="span--icon__right-margin" />
+                  Filter
+                </DropdownToggle>
+              }
+            />
+          </div>
+        )}
+        <div className="pf-c-input-group co-search-group__filters co-search-group__resource">
+          {!hideLabelFilter && (
+            <DropdownInternal
+              items={FilterType}
+              onChange={switchFilter}
+              selectedKey={filterType}
+              title={filterType}
+            />
+          )}
+          {!hideNameFilter && (
+            <AutocompleteInput
+              className="co-text-node"
+              onSuggestionSelect={(selected) => {
+                updateLabelFilter([...labelFilters, selected]);
+              }}
+              showSuggestions={FilterType.LABEL === filterType}
+              // Text Input value
```

Comment:
nit: This should be apparent... I don't see a lot of value in this comment or the one just below on line 256.